client/client.go

```diff
@@ -35,6 +37,29 @@ var (
 	errorAddingServerCertificateToPool = errors.New("Error adding trusted server certificate to pool.")
 )
 
+var (
+	// 	                  http.Client.Timeout
+	// +--------------------------------------------------------+
+	// +--------+  +---------+  +-------+  +--------+  +--------+
+	// |  Dial  |  |   TLS   |  |Request|  |Response|  |Response|
+	// |        |  |handshake|  |       |  |headers |  |body    |
+	// +--------+  +---------+  +-------+  +--------+  +--------+
+	// +--------+  +---------+             +--------+
+	//  Dial        TLS                     Response
+	//  timeout     handshake               header
+	//  timeout                             timeout
+	//
+	//  It covers the entire exchange, from Dial (if a connection is not reused)
+	// to reading the body. This is to timeout long lasing connections.
+	//
+	// 4 hours shold be enough to download 2GB image file with the
+	// average download spead ~1 mbps
+	defaultClientReadingTimeout = 4 * time.Hour
+
+	// connection keepalive options
+	connectionKeepaliveTime = 10 * time.Second
```

Comment:
I think 10 seconds is perhaps a tiny bit short. Links like GSM may be really crappy, and even my local wireless may sometimes exceed ten seconds if it changes channel for example. Maybe sixty seconds?